src/interactivity/win32/uiaTextRange.cpp

```diff
@@ -10,12 +10,15 @@
 
 using namespace Microsoft::Console::Types;
 using namespace Microsoft::Console::Interactivity::Win32;
+using namespace Microsoft::WRL;
 using Microsoft::Console::Interactivity::ServiceLocator;
 
-std::deque<UiaTextRange*> UiaTextRange::GetSelectionRanges(_In_ IUiaData* pData,
-                                                           _In_ IRawElementProviderSimple* pProvider)
+HRESULT UiaTextRange::GetSelectionRanges(_In_ IUiaData* pData,
+                                         _In_ IRawElementProviderSimple* pProvider,
+                                         _Out_ std::deque<ComPtr<UiaTextRange>>& ranges)
+try

```

Comment:
woah is this a pattern we've used anywhere else, with the entire function just being a `try{ ... } CATCH_RETURN`? I'd think that stylistically we'd still wrap that in another set of braces